### ISSUE TITLE
LSeg on ScanNet

### ISSUE THREAD

**Issue by HustQBW** [pengsongyou/openscene#9]
Hello, @pengsongyou ,thanks for your great job. I've done the scannet lseg feature extraction. The code is :
`def extract_lseg_img_feature(img_dir, lseg_model, text_emb,img_size=None, regional_pool=True):
    '''Extract per-pixel OpenSeg features.'''

    # load RGB image
    lseg_model = lseg_model.cuda()
    lseg_model.eval()
    np_img = cv2.imread(img_dir)
    h,w,_ = np_img.shape
    new_w,new_h = 320,256 # (320,240) can't divide 32

    padded_image = cv2.copyMakeBorder(np_img, 0, new_h - h, 0, new_w - w, cv2.BORDER_CONSTANT, value=0)
    input_img = transforms.ToTensor()(padded_image)
    input_img = transforms.Normalize(mean=IMAGENET_DEFAULT_MEAN,std=IMAGENET_DEFAULT_STD)(input_img)
    input_img = input_img.unsqueeze(dim=0)
    
    # run LSeg
    with torch.no_grad():
        feat_2d = lseg_model(input_img.cuda())
    feat_2d = feat_2d[:,:,:h,:w].squeeze(dim=0)
    feat_2d = feat_2d.detach().half().cpu() # 不加detach的话，这些计算图不会释放，显存会累加，导致爆炸

    return feat_2d`

In this way, I use the demo model :
<img width="507" alt="image" src="https://user-images.githubusercontent.com/48615375/233350233-23363467-b395-4671-a46b-39ac4b4359be.png">
But I only get 31 mIoU on ScanNet. Instead, your multi-view lseg feature get 49 mIoU. So I am really curious about your implement here. Can you point out the mistake on my code? Thanks for your reply!

**Comment by yhyang-myron** [comment]
Hi, @pengsongyou, I also have the similar question. Did you use multi-scale eval in Lseg to extract features?

**Comment by pengsongyou** [comment]
@HustQBW please refer to my reply in https://github.com/pengsongyou/openscene/issues/8 for the fusion code.

@yhyang-myron I only use the single scale LSeg to extract features. For the crop size I need to further check my code as well, which I should do it within the coming week. Please stay tuned a bit.

Best
Songyou

**Comment by Wang-pengfei** [comment]
Hi, @pengsongyou. If you have time, could you please organize the code for Lseg? I am also facing the crop_size and other issues mentioned above.
